### PR TITLE
[Bugfix] Immunities are again shown. 

### LINKED ISSUE #3573: [Bug] Missing type-hints for immunities
## Describe the bug

See screenshots.
Type-hints don't display correctly

## Expected behavior

the type should be "grayed out" since it doesn't affect the enemy pkm (unless some special conditions are met)

## Screenshots / Videos

image

### Additional context

Discord Messages:
- https://discord.com/channels/1125469663833370665/1273029028210671748/1273821780175229021
- https://discord.com/channels/1125469663833370665/1176874654015684739/1273963881718616076
- https://discord.com/channels/1125469663833370665/1176874654015684739/1273964747804377088
- [possible fix - by allen](https://discord.com/channels/1125469663833370665/1176874654015684739/1273994191491563521)

### PR DESCRIPTION
**Pull request by CodeTappert**
(While keeping the null check) 
Fixes #3573

## What are the changes the user will see?

Moves that are 0x effective (immunities) will again be shown as greyed out

## Why am I making these changes?

Because the not null change by @flx-sta made them look like 1x moves

## What are the changes from a developer perspective?

I changed it so it wont filter out 0 but everything else that was filtered out by flxs method (undefined, "", null etc)

### Screenshots/Videos

https://github.com/user-attachments/assets/c5e8bdbf-357a-4202-a863-a3dd69a89289

## How to test the changes?

Have a pokemon have a move that is 0x effective to the opponent

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I add placeholders for them in locales?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by flx-sta** [comment]
I actually have a fix ready. I just got caught in writing test coverage for it that’s why I haven’t opened it yet

**Comment by flx-sta** [comment]
The proper fix for me in the end was to change the “get Move effectiveness” method to not return “undefined” on status moves but “1” instead.

File: src/ui/fight-ui-handler.ts [hunk 1]
```diff
@@ -273,7 +273,9 @@ export default class FightUiHandler extends UiHandler {
 
     const moveColors = opponents.map((opponent) => {
       return opponent.getMoveEffectiveness(pokemon, pokemonMove);
-    }).filter((eff) => !!eff).sort((a, b) => b - a).map((effectiveness) => {
+    }).filter((eff): eff is TypeDamageMultiplier => {
+      return true;
+    }).sort((a, b) => (b ?? 0) - (a ?? 0)).map((effectiveness) => {
```
Comment (flx-sta) [inline src/ui/fight-ui-handler.ts:278]: Did you check what a “status” move is showing ? (E.g. growl)

**Comment by Snailman11** [comment]
Link to Discord
https://discord.com/channels/1125469663833370665/1274438543959392276

File: src/ui/fight-ui-handler.ts (hunk 1 above)
Comment (CodeTappert) [reply src/ui/fight-ui-handler.ts:278]: In my video you can see "Disable" which counts as a status move

**Pull request closed without merging**